examples/run_CPMG.py

```diff
@@ -31,31 +30,25 @@
 # {{{set phase cycling
 phase_cycling = True
 if phase_cycling:
-    ph1_cyc = r_[0, 1, 2, 3]
-    nPhaseSteps = 4
+    ph1_cyc = array([(j*2+k)%4 for k in range(4) for j in range(2)])
+    ph2_cyc = array([(k+1)%4 for k in range(4) for j in range(2)])
+    nPhaseSteps = 8
+
 if not phase_cycling:
     ph1_cyc = 0.0
     nPhaseSteps = 1
 # }}}
-# {{{better tau
-marker = 1.0  # 10/10/22 → what is this? → pretty sure the time needed to execute the marker command
-pad_start = config_dict["tau_extra_us"] - config_dict["deadtime_us"]
-pad_end = (
-    config_dict["tau_extra_us"] - config_dict["deblank_us"] - marker
-)  # marker + deblank
-assert(pad_start > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-assert (pad_end > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-twice_tau_echo_us = (  # the period between 180 pulses
-    config_dict["tau_extra_us"] * 2 + config_dict["acq_time_ms"] * 1e3
+# {{{symmetric tau
+marker = 1.0
+jumpto = 1.0
+pad_end = config_dict['deadtime_us'] - config_dict['deblank_us'] - marker - jumpto
+twice_tau_echo_us = (  # the period between end of first 180 pulse and start of next
+    config_dict['deadtime_us'] + config_dict["acq_time_ms"] * 1e3 + pad_end + marker + config_dict['deblank_us']
 )
-# now twice_tau_echo_us/2.0 is τ_echo, so I need to subtract the extra delays
-# imposed by the ppg to determine the remaining τ that I need
-config_dict["tau_us"] = twice_tau_echo_us / 2.0 - (
+config_dict["tau_us"] = twice_tau_echo_us / 2.0 + (
```

Comment:
this doesn't work.  You should define evolution during the pulse before this, and subtract it here.  Also, as noted w.r.t. the diagram, are you sure there is no delay either after the 90 or before the 180 that you need to include here?

I'm stopping here -- this PR is not ready.

Reply:
(1) twice_tau echo does not include the tau_evol so it needs to be added
(2) there is no deadtime involved with the tau only a deblank prior to the 180 pulse which I added in as well

Reply:
done and discussed